Server/JokesOnYou.Web.Api/Repositories/UserRepository.cs

```diff
@@ -7,22 +7,33 @@
 using JokesOnYou.Web.Api.Repositories.Interfaces;
 using Microsoft.AspNetCore.Identity;
 using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using AutoMapper;
 
 namespace JokesOnYou.Web.Api.Repositories
 {
     public class UserRepository : IUserRepository
     {
         private readonly UserManager<User> _userManager;
+        readonly IMapper _mapper;
 
-        public UserRepository(UserManager<User> userManager)
+        public UserRepository(UserManager<User> userManager, IMapper mapper)
         {
+            _mapper = mapper;
             _userManager = userManager;
         }
 
         public async Task<User> CreateUserAsync(UserRegisterDTO userRegisterDTO)
         {
-            var userCreationResult = await _userManager.CreateAsync(new User() { Email = userRegisterDTO.Email, UserName = userRegisterDTO.Email }, userRegisterDTO.Password);
+            if (_userManager.Users.Any(x => x.Email == userRegisterDTO.Email))
+                throw new Exception($"Email {userRegisterDTO.Email} is already taken");
 
+            if (_userManager.Users.Any(x => x.UserName == userRegisterDTO.UserName))
```

Comment:
this check is default in Identity:
https://stackoverflow.com/questions/23028317/asp-net-identity-username-is-unique/23252357